src/main/src/main/java/org/geoserver/catalog/CoverageViewReader.java

```diff
@@ -480,6 +484,60 @@ public GridCoverage2D read(GeneralParameterValue[] parameters)
         return result;
     }
 
+    /**
+     * The BandMerge operation takes indexed images and expands them, however in the context of
+     * coverage view band merging we don't normally want that, e.g., raster mask bands are
+     * represented as indexed but we really want to keep them in their binary, single band form. To
+     * do so, the IndexColorModel is replaced by a ComponentColorModel
+     *
+     * @param coverage
+     * @return
+     */
+    private GridCoverage2D prepareForBandMerge(GridCoverage2D coverage) {
+        RenderedImage ri = coverage.getRenderedImage();
+        SampleModel sampleModel = ri.getSampleModel();
+        if (sampleModel.getNumBands() == 1 && ri.getColorModel() instanceof IndexColorModel) {
+            // format with same data type as input
+            final ParameterBlock paramBlk = new ParameterBlock().addSource(ri);
+            int dataType = sampleModel.getDataType();
+            paramBlk.add(dataType);
+            // force the desired color model
+            ComponentColorModel targetColorModel =
+                    new ComponentColorModel(
+                            ColorSpace.getInstance(ColorSpace.CS_GRAY),
+                            false,
+                            false,
+                            Transparency.OPAQUE,
+                            dataType);
+            ImageLayout layout = new ImageLayout(ri);
+            layout.setColorModel(targetColorModel);
+
+            // that might not be enough, the source sample model might not be compatible
+            // e.g., MultiPixelPackedSampleModel is not compatible with ComponentColorModel
+            if (!targetColorModel.isCompatibleSampleModel(sampleModel)) {
+                SampleModel targetSampleModel =
+                        targetColorModel.createCompatibleSampleModel(
+                                sampleModel.getWidth(), sampleModel.getHeight());
+                layout.setSampleModel(targetSampleModel);
+            }
+
+            Hints localHints = new Hints(hints);
+            localHints.put(JAI.KEY_IMAGE_LAYOUT, layout);
+            RenderedOp formatted = JAI.create("Format", paramBlk, localHints);
+
+            return new GridCoverageFactory()
+                    .create(
+                            coverage.getName(),
+                            formatted,
+                            coverage.getGridGeometry(),
+                            coverage.getSampleDimensions(),
```

Comment:
Are you sure this is correct, I mean, are we suere the original coverage has 3 SD and not 1.
Just checking.

Reply:
The branch starts with "if(sampleModel.getNumBands() == 1" so the source image should have one sample dimension, just like the output one.